logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/team/SyncSelfTeamUseCase.kt

```diff
@@ -0,0 +1,22 @@
+package com.wire.kalium.logic.feature.team
+
+import com.wire.kalium.logic.CoreFailure
+import com.wire.kalium.logic.data.team.TeamRepository
+import com.wire.kalium.logic.data.user.UserRepository
+import com.wire.kalium.logic.functional.Either
+import kotlinx.coroutines.flow.first
+
+interface SyncSelfTeamUseCase {
+    suspend operator fun invoke(): Either<CoreFailure, Unit>
+}
+
+class SyncSelfTeamUseCaseImpl(
+    private val userRepository: UserRepository,
+    private val teamRepository: TeamRepository
+) : SyncSelfTeamUseCase {
+
+    override suspend fun invoke(): Either<CoreFailure, Unit> =
+        userRepository.getSelfUser().first().team?.let { teamId ->
+            teamRepository.fetchTeamById(teamId = teamId)
+        } ?: Either.Right(Unit)
```

Comment:
Fail silently?

_AT LEAST_ a warning log.

This could be an `UnknownFailure(IllegalStateException("SyncSelfTeam cannot be called before SyncSelfUser has succeeded")`

Reply:
Actually a user could not have a team, thus just keep the flow going, if there is an error on fetching the user team, then an exception is thrown.

Otherwise if a User doesn't have a team, then we continue the flow as it should normally (following to syncing contacts).

Does it makes sense? That's how I read the SyncFlow we are having based on users that have and do not have a team.

Reply:
Oh shoot!

Nevermind my past comment. I got it wrong :P 

I was thinking about the possibility of `getSelfUser()` returning null, but it returns a `Flow`. So this won't happen.

In any case, I guess a logging saying `Skipping team sync because user doesn't belong to a team` would be a good to have.

Reply:
🙈  Sure! I will be adding it in the next PR as I have already added some more things to this UseCase hehe